### ISSUE TITLE
Tried calling: onRequest(Instance of 'RequestOptions'), data: null}

### ISSUE THREAD

**Issue by baoxiehao** [flutterchina/dio#319]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter test   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   2.1.6   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Run dio http request in flutter test, then it will report no network error.

handleError(): response = HttpResponse{**requestCode: -1**, serviceCode: 0, msg: DioError [DioErrorType.DEFAULT]: NoSuchMethodError: The method 'onRequest' was called on null.
Receiver: null
Tried calling: onRequest(Instance of 'RequestOptions'), data: null}

**Comment by wendux** [comment]
Some code?
